sgkit/stats/utils.py

```diff
@@ -0,0 +1,102 @@
+from typing import Hashable, Tuple
+
+import numpy as np
+import xarray as xr
+from dask.array import Array
+from xarray import DataArray, Dataset
+
+from ..typing import ArrayLike
+
+
+def concat_2d(ds: Dataset, dims: Tuple[Hashable, Hashable]) -> DataArray:
+    """Concatenate dataset with a mixture of <= 2D variables as single DataArray.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Dataset containing variables to convert.
+        Any variables with a first dimension not equal to `dims[0]`
+        will be ignored.
+    dims : Tuple[Hashable, Hashable]
+        Names of resulting dimensions in 2D array where first dimension
+        is shared by all variables and all others are collapsed into
+        a new dimension named by the second item.
+
+    Returns
+    -------
+    DataArray
+        Array with dimensions defined by `dims`.
+    """
+    arrs = []
+    for var in ds:
+        arr = ds[var]
+        if arr.dims[0] != dims[0]:
```

Comment:
where is this condition tested?

Reply:
By this line: https://github.com/pystatgen/sgkit/blob/eac/stats_utils/sgkit/tests/test_stats_utils.py#L106

The first dimensions are assigned different labels there.